### ISSUE TITLE
Do NOT add domain to list when not being created successfully

### ISSUE THREAD

**Issue by FibreFoX** [Neilpang/acme.sh#997]
Background
-----------------
While developing a simple ansible-role for acme.sh, I encountered this problem. Im developing using some small local VM (using vagrant+ansible) but it mostly has no connection to the internet. After having acme.sh installed while internet connection was available, I experimented a bit with the client.

Steps to reproduce
------------------
* cut internet connection
* execute `acme.sh --issue -d SOME-NOT-YET-REGISTERED-DOMAIN.TLD --webroot /var/www/SOMEWHERE` (doesn't matter if `--test` was provided) (this outputs some warnings and fails)
* execute `acme.sh --list --no-color`

Expected
------------------
Domain should not be existing inside that list, or flagged as "not created"

Actual result
------------------
Domain gets listed

**Comment by Takios** [comment]
There is no date in the "Created" column when the issuance fails for some reason.

**Comment by FibreFoX** [comment]
Hm, for automation-purpose, would it be possible to add some CLI command where this can be returned in a proper way? Maybe the `--listraw` output can be enhanced a bit. As there is no documentation about this, I thought this to be a bug, and it does not feel good IMHO. What do you think about this?

**Comment by Takios** [comment]
I agree that it does feel a bit wonky. The benefit is, that commands like "--renew-all" will retry to issue the certificate. The drawback is that if it's a permanent error (like a non-existing/mistyped domain), it'll drain resources and might put one above a rate limit faster than expected.

For automation, I think checking the Created column should be sufficient. Also, you can check the return code of acme.sh when issuing the certificate, it'll be 1 if it's unsuccessful. You can then act on it accordingly.

But I don't want to speak for @Neilpang, maybe he's already got some other plans for this?

**Comment by FibreFoX** [comment]
I'm a fan of concrete status/labels, something like "active, removed, pending, failed" and so on, but having to interpret the shown data gets a bit of a gamble game, because every false interpretation makes troubleshooting even worse.

The list returned by `--list` is a mixed list with "not yet issued certificate" and "existing/active certificate" and "existing/active certificate from testing/staging server" (when having `--test` used while creating).

Maybe this is all intended, but some documentation is required for this (at least would help people like me understanding what is happening).

**Comment by Neilpang** [comment]
Yes,  I know the doc for the `--list` command is not enough.
but nobody complains about it since I added it.   so, I think there are few people using it.
That's why I didn't take much care on it.

If you have any advice,  you are free to send a PR, I will think about it.

You can also tell me what you need/want, I will try my best to implement.

thanks.

**Comment by FibreFoX** [comment]
Thanks for this positive feedback. I will think about all my requirements, maybe even create a pull-request if possible. Will head back to this isse later.

**Comment by neilpang** [comment]
Behavior is intentional: the domain conf is written at the start of `--issue` so that the attempted parameters are preserved and a later `--renew` / cron run can retry exactly what you asked for. If an experiment should not stay on the renewal list, remove it with `acme.sh --remove -d example.com` (and for throwaway experiments, a separate `--config-home` keeps your real config clean). Closing.
